File: src/app/model/statement.ts
import { Appointment } from "./appointment";

export class Statement {
    bloodTypeName: any='';
    napomena: string = '';
    bakarSulfat: string = '';
    hemoglobinometar: string = '';
    srce: string = '';
    pluca: string = '';
    krvniPritisak: string = '';
    zahtev: boolean = false;
    napomenaPregleda: string = '';
    razlogOdbijanja: string = '';
    mestoPunkcije: string = '';
    kolicinaUzeteKrvi: string = '';
    pocetakKrvi: string = '';
    krajKrvi: string = '';
    appointmentId: number;
    // appointments: Appointment | null = null;
    // historyOfRegisterUser: HistoryOfRegisterUser | null = null;
  
    constructor(obj?: any) {
      if (obj) {

        this.bloodTypeName = obj.bloodTypeName;
        this.napomena = obj.napomena;
        this.bakarSulfat = obj.bakarSulfat;
        this.hemoglobinometar = obj.hemoglobinometar;
        this.srce = obj.srce;
        this.pluca = obj.pluca;
        this.krvniPritisak = obj.krvniPritisak;
        this.zahtev = obj.zahtev;
        this.napomenaPregleda = obj.napomenaPregleda;
        this.razlogOdbijanja = obj.razlogOdbijanja;
        this.mestoPunkcije = obj.mestoPunkcije;
        this.kolicinaUzeteKrvi = obj.kolicinaUzeteKrvi;
        this.pocetakKrvi = obj.pocetakKrvi;
        this.krajKrvi = obj.krajKrvi;
        this.appointmentId = obj.appointmentId;
        // this.appointments = obj.appointments;
        // this.historyOfRegisterUser = obj.historyOfRegisterUser;
      }
    }
  }

File: src/main/java/isa/projekat/projektniZadatak/model/Statement.java
package isa.projekat.projektniZadatak.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonIncludeProperties;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@NoArgsConstructor
@Getter
@Setter
public class Statement {

    @Id
    @SequenceGenerator(name = "statementSeqGen", sequenceName = "statementSeqGen", initialValue = 1, allocationSize = 1)
    @GeneratedValue(strategy = GenerationType.IDENTITY, generator = "statementSeqGen")
    @Column(name="id", unique=true, nullable=false)
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER)
    private BloodType bloodType;

    private String napomena;

    private String bakarSulfat;

    private String hemoglobinometar;

    private String srce;

    private String pluca;

    private String krvniPritisak;

    private boolean zahtev;

    private String napomenaPregleda;

    private String razlogOdbijanja;

    private String mestoPunkcije;

    private String kolicinaUzeteKrvi;

    private String pocetakKrvi;

    private String krajKrvi;


    @OneToOne(mappedBy = "statement", fetch = FetchType.EAGER)
    @JsonIgnoreProperties("statement")
    private Appointments appointments;

    @OneToOne(mappedBy = "statement", fetch = FetchType.EAGER)
    @JsonIgnoreProperties("statement")
    private HistoryOfRegisterUser historyOfRegisterUser;



}
